### PR TITLE
[SofaSphFluid] Reorder plugin code and scenes files

### PR DESCRIPTION
**Pull request by epernod**
______________________________________________________

This PR: 
- [x] builds with SUCCESS for all platforms on the CI.
- [x] does not generate new warnings.
- [x] does not generate new unit test failures.
- [x] does not generate new scene test failures.
- [x] does not break API compatibility.
- [x] is more than 1 week old (or has fast-merge label).

**Reviewers will merge only if all these checks are true.**

### CODE REVIEW

File: applications/plugins/SofaSphFluid/src/SofaSphFluid/config.h.in [hunk 1]
```diff
@@ -22,10 +22,9 @@
 #ifndef SOFASPHFLUID_CONFIG_H
 #define SOFASPHFLUID_CONFIG_H
 
-#include <SofaBase/config.h>
+#include <sofa/helper/system/config.h>
 
 #ifdef SOFA_BUILD_SPH_FLUID
-#  define SOFA_TARGET SofaSphFluid
```
Comment (guparan) [inline applications/plugins/SofaSphFluid/src/SofaSphFluid/config.h.in:28]: This should be kept :)

**Review by guparan — CHANGES_REQUESTED** [review]

**Comment by damienmarchal** [comment]
Hello @epernod,

For the future: 
Shouldn't we move the SpatialGridContainer into SofaVolumetricData ? In addition there is in SofaEulerianFluid some dedicated data structure for storing the quantities. 
Merging al that could be intereting. 

Shouldn't we merge everything into a single SofaFluids plugins ? 

EDIT: And the MarchingCube code is also duplicated in several places
EDIT2: And the FastMarching algorithm either :)

**Comment by epernod** [comment]
@damienmarchal yes totally agree on the different points. But as you say: "for the future".
I will move one step at a time, by making this plugin works again first.

But indeed it is a topic I wanted to speak about but the meeting was already too long.
It would be good to either merge some plugins or have some high level option in the cmake of sofa, for example, SOFA_HAVE_FLUID which will add, Sph, Eulerian, Volumetric rendering plugins etc...

**Comment by epernod** [comment]
@damienmarchal it could be useful (for me at least) to open issue(s) to point where there are duplicated codes or same algo needed.

**Comment by fredroy** [comment]
[ci-build][with-regression-tests][with-scene-tests]

**Review by guparan — APPROVED** [review]

**Pull request merged**